server/src/main/java/org/opensearch/search/pipeline/Pipeline.java

```diff
@@ -146,11 +147,18 @@ void transformRequest(SearchRequest request, ActionListener<SearchRequest> reque
             final ActionListener<SearchRequest> nextListener = currentListener;
             SearchRequestProcessor processor = searchRequestProcessors.get(i);
             currentListener = ActionListener.wrap(r -> {
+                ProcessorExecutionDetail detail = new ProcessorExecutionDetail(processor.getType());
+                detail.addInput(r.source().shallowCopy());
```

Comment:
can you add check for `verbosePipeline` here, we're adding unnecessary overhead of doing shallow copy in case verbose is not enabled